planner/core/preprocess.go

```diff
@@ -1592,11 +1592,12 @@ func (p *preprocessor) handleAsOfAndReadTS(node *ast.AsOfClause) {
 		return
 	}
 	if p.LastSnapshotTS != 0 {
-		dom := domain.GetDomain(p.ctx)
-		p.InfoSchema, p.err = dom.GetSnapshotInfoSchema(p.LastSnapshotTS)
+		is, err := p.ctx.GetSnapshotInfoSchema(p.LastSnapshotTS)
+		p.err = err
 		if p.err != nil {
 			return
 		}
+		p.InfoSchema = is.(infoschema.InfoSchema)
```

Comment:
What will happen if `is` is not a `infoschema.InfoSchema`
`p.InfoSchema` is `nil` and no error set, than maybe the code panic in some other place?